Godot 4.4 improvements

Joy-less opened the pull request:
There's a few things going on here:
- Added type annotations
- Renamed `hover` signal to `hovered` to match `selected`
- Added `center_offset` property to move the icons further from / closer to the center
- Other improvements

Joy-less committed:
Godot 4.4 improvements

tavurth commented:
Looks good and clean, I will try to test it out a little later today 👍

tavurth commented:
I notice there's an off-by-one index mistake on the radial selection? Are you seeing that on your version?

### Current 4.x version:

<img width="688" alt="Screenshot 2025-06-15 at 19 29 31" src="https://github.com/user-attachments/assets/f57f8e59-38fd-4cb8-9f80-76b52bdbb821" />

###  This PR version:
<img width="688" alt="Screenshot 2025-06-15 at 19 30 15" src="https://github.com/user-attachments/assets/24f7cbea-c9bf-4d9e-8e30-c25da7164acc" />

Joy-less committed:
Fixes & more improvements

Joy-less commented:
@tavurth
Hi, I've managed to trace the issue.

The reason is that I replaced `len(self.get_children())` with `self.get_child_count()`. But the latter actually returns an extra child.

Code:
```gd
func _init() -> void:
	print("INIT")
	
	print("get_children: ", self.get_children())
	print("len get_children: ", len(self.get_children()))
	print("get_child_count: ", self.get_child_count())
	
	self.add_child(preload("./RadialMenu.tscn").instantiate())
	
	print("get_children: ", self.get_children())
	print("len get_children: ", len(self.get_children()))
	print("get_child_count: ", self.get_child_count())
```
Output:
```
get_children: []
len get_children: 0
get_child_count: 0
get_children: []
len get_children: 0
get_child_count: 1
```

After a while of debugging I found what the issue is. `self.get_children()` is ignoring the `RadialMenu.tscn` whereas `self.get_child_count()` is not. The reason is that `self.get_children()` is overriden to remove `RadialMenu.tscn`.

I removed this override and changed `RadialMenu.tscn` to use `Node.INTERNAL_MODE_FRONT` which hides it from both of these methods and is simpler.

tavurth commented:
LGTM 👍 

Thank you!

tavurth commented:
Updated asset in library
